src/cli/Microsoft.DotNet.UpgradeAssistant.Cli/Commands/Upgrade/ConsoleUpgrade.cs

```diff
@@ -76,9 +86,32 @@ public async Task RunAsync(CancellationToken token)
             {
                 // Do not pass the same token as it may have been canceled and we still need to persist this.
                 await _stateManager.SaveStateAsync(context, default);
+                await WriteUpgradeReport(context, token);
             }
         }
 
+        private async Task<bool> WriteUpgradeReport(IUpgradeContext context, CancellationToken token)
+        {
+            var outputDirectory = Path.GetDirectoryName(context.InputPath);
+            if (outputDirectory is null || !context.Results.Any())
```

Comment:
I think I'd find value in an empty report if only to say "No more steps"

Reply:
This will need modifying all integration tests with such file. I'll do that anyway as a part of the next PR where I actually use this to add to results. Will take care of this then.